spec/components/email/receiver_spec.rb

```diff
@@ -651,6 +670,14 @@ def process(email_name)
       SiteSetting.ignore_by_title = "foo"
       expect { process(:ignored) }.to_not change(Topic, :count)
     end
+
+    it "associates email from a secondary address with user" do
+      user = Fabricate(:user, trust_level: SiteSetting.email_in_min_trust)
+      Fabricate(:secondary_email, email: "existing@bar.com", user: user)
+
+      expect { process(:existing_user) }.to change(Topic, :count)
+      expect(Topic.last.user.id).to eq user.id
```

Comment:
We need to test for the content of the topic here as well. Let's rewrite the assertion to `expect(Topic.last.user).to eq(user)`.